From d53238b614e01266a3d36b417b60a502e0698504 Mon Sep 17 00:00:00 2001
From: Chao Yu <chao@kernel.org>
Date: Mon, 7 Jul 2025 16:48:32 +0800
Subject: erofs: fix to add missing tracepoint in erofs_readahead()

From: Chao Yu <chao@kernel.org>

commit d53238b614e01266a3d36b417b60a502e0698504 upstream.

Commit 771c994ea51f ("erofs: convert all uncompressed cases to iomap")
converts to use iomap interface, it removed trace_erofs_readahead()
tracepoint in the meantime, let's add it back.

Fixes: 771c994ea51f ("erofs: convert all uncompressed cases to iomap")
Signed-off-by: Chao Yu <chao@kernel.org>
Reviewed-by: Gao Xiang <hsiangkao@linux.alibaba.com>
Link: https://lore.kernel.org/r/20250707084832.2725677-1-chao@kernel.org
Signed-off-by: Gao Xiang <hsiangkao@linux.alibaba.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 fs/erofs/data.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/erofs/data.c
+++ b/fs/erofs/data.c
@@ -383,6 +383,9 @@ static int erofs_read_folio(struct file
 
 static void erofs_readahead(struct readahead_control *rac)
 {
+	trace_erofs_readahead(rac->mapping->host, readahead_index(rac),
+					readahead_count(rac), true);
+
 	return iomap_readahead(rac, &erofs_iomap_ops);
 }
 
